src/media/inner.rs

```diff
@@ -746,58 +759,30 @@ impl MediaInner {
-            // TODO: This function should be split into two halves, but because of the borrow checker
-            // it's hard to construct.
-
-            // This first scope tries to send a spurious (unasked for) resend of a packet already sent.
-            if pad_size > MIN_SPURIOUS_PADDING_SIZE {
-                let pt = self
-                    .buffers_tx
-                    .iter()
-                    .map(|(pt, buffer)| (pt, buffer.history_size()))
-                    .filter(|(_, size)| *size > 0)
-                    // Use the last PT i.e. the one that has the most RTX history, this is a poor
-                    // approximation for most recent sends.
-                    .max_by_key(|(_, size)| *size)
-                    .map(|(pt, _)| *pt);
-
-                // If we find a pt above, we do a spurious (unasked for) resend of this packet.
-                if let Some(pt) = pt {
-                    let buffer = get_buffer_tx(&self.buffers_tx, pt)
-                        .expect("the buffer to exist as verified previously");
-
-                    // Find a historic packet that is smaller than this max size. The max size
-                    // is a headroom since we can accept slightly larger padding than asked for.
-                    let max_size = pad_size * 2;
-                    if let Some(packet) = buffer.historic_packet_smaller_than(max_size) {
-                        let seq_no = packet.seq_no.expect(
-                            "this packet to have been sent and therefore have a sequence number",
-                        );
-                        // Saturating sub because we can overflow and want to stop when that
-                        // happens.
-                        pad_size = pad_size.saturating_sub(packet.data.len());
-                        trace!(
-                            "Queued {} bytes worth of resend padding, seq_no = {seq_no}",
-                            packet.data.len()
-                        );
-
-                        self.padding.push_back(Padding::Spurious(Resend {
-                            pt,
-                            body_size: packet.data.len(),
-                            ssrc: packet.meta.ssrc,
-                            seq_no,
-                            queued_at: now,
-                        }));
-
-                        continue;
-                    }
-                }
-            }
+            if let Some((pt, packet)) = self.spurious_resend_candidate(pad_size) {
+                // Find a historic packet that is smaller than this max size. The max size
+                // is a headroom since we can accept slightly larger padding than asked for.
+                let seq_no = packet
+                    .seq_no
+                    .expect("this packet to have been sent and therefore have a sequence number");
+                // Saturating sub because we can overflow and want to stop when that
+                // happens.
```

Comment:
I think this needs to say _why_ it can overflow.

Reply:
This is an old comment of mine, it's because we allow providing a packet larger than the request padding i.e. you can have 491 bytes of padding outstanding that is satisfied by a packet that's 600 bytes large.

Reply:
but because i moved the lines above related to picking the size, it was no longer in context. Now i adjusted